sql/hive/src/test/scala/org/apache/spark/sql/hive/MetastoreDataSourcesSuite.scala

```diff
@@ -528,7 +528,7 @@ class MetastoreDataSourcesSuite extends QueryTest
           val e = intercept[AnalysisException] {
               sparkSession.catalog.createTable("createdJsonTable", jsonFilePath.toString)
             }
-          checkErrorTableAlreadyExists(e, s"`$SESSION_CATALOG_NAME`.`default`.`createdJsonTable`")
+          checkErrorTableAlreadyExists(e, s"`$SESSION_CATALOG_NAME`.`default`.`createdjsontable`")
```

Comment:
This makes me worry a little about case-sensitive case. To make it sure, could you make this `test("create external table")` to run both case-sensitive and case-insensitive cases by adding `withSQLConf(SQLConf.CASE_SENSITIVE.key -> `?

Reply:
There was a PR from @cloud-fan to centralize qualifying table identifiers which will `format` tables by the case-sensitivity setting. But it seemed missing the `sparkSession.catalog.createTable` API.

But I'm OK to enhance the test here with you.